serializers/base.rb

```diff
@@ -24,33 +24,13 @@
 
 module Serializers
   class Base
-    @@structures = {}
-
-    def self.structure(type, &blk)
-      @@structures["#{name}::#{type}"] = blk
-    end
-
-    def self.serialize(object)
-      new.serialize(object)
-    end
-
-    def initialize(type = :default)
-      @type = type
-    end
-
-    def serialize(object)
+    def self.serialize(object, options = {})
       return if object.nil?
       if object.respond_to?(:map) && !object.is_a?(Hash)
-        object.map { |item| serializer.call(item) }
+        object.map { |item| serialize_internal(item, options) }
       else
-        serializer.call(object)
+        serialize_internal(object, options)
       end
     end
-
```

Comment:
Should we add a `serialize_internal` stub to the parent class to indicate that it needs to be implemented in the child classes? I searched for `serialize_internal` in the current `serializers/base.rb` file at the beginning, then I realized it's only in the children.

```ruby
    def self.serialize_internal(object, options = {})
      raise NoMethodError
    end
```